### PR TITLE
[usage] Allow selecting billed months

### LINKED ISSUE #11526: Allow users to switch between months and update usage page sidebar designs
### Problem to solve

Following up from https://github.com/gitpod-io/gitpod/pull/11481, the sidebar in the new usage page does not support switching between other periods (months).

### Proposal

| Sidebar | Sidebar (Dropdown) |
|-|-|
| Sidebar-1 | Sidebar |

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Allows users to select billing months.

Screenshot 2022-07-27 at 10 43 37

## Related Issue(s)

Fixes #11526

## How to test
Go to https://lau-time-view-11526.preview.gitpod-dev.com/t/gitpod/usage and click on the previous months.

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/teams/TeamUsage.tsx [hunk 1]
```diff
@@ -90,50 +118,29 @@ function TeamUsage() {
...
-                                                    y2="17.107"
-                                                    gradientUnits="userSpaceOnUse"
-                                                >
-                                                    <stop stop-color="#FFAD33" />
-                                                    <stop offset="1" stop-color="#FF8A00" />
-                                                </linearGradient>
-                                            </defs>
-                                        </svg>
-                                        <span>{calculateTotalUsage()} Total Credits</span>
+                                        <CreditsSvg className="my-auto mr-1" />
```
Comment (laushinka) [inline components/dashboard/src/teams/TeamUsage.tsx:142]: This is a drive-by improvement to extract the SVGs.

File: components/dashboard/src/teams/TeamUsage.tsx [hunk 2]
```diff
@@ -82,6 +84,33 @@ function TeamUsage() {
...
+        setEndDateOfBillMonth(end);
+    };
+
+    const getBillingHistory = () => {
+        let rows = [];
+        // This goes back 6 months from the current month
+        for (let i = 1; i < 7; i++) {
+            const endDateVar = i - 1;
+            const startDate = new Date(today.getFullYear(), today.getMonth() - i);
+            const endDate = new Date(today.getFullYear(), today.getMonth() - endDateVar, 0);
```
Comment (easyCZ) [inline components/dashboard/src/teams/TeamUsage.tsx:98]: Using this might make it easier logic vise. https://stackoverflow.com/questions/41505492/how-to-subtract-one-month-using-moment-js

I suspect the current logic has a bug for when 7 months back goes back to the previous year, as then you need to wrap the month and subtract 1 year

Comment (easyCZ) [reply components/dashboard/src/teams/TeamUsage.tsx:98]: You can test it by changing it to generate 12 months, or even 24 to see it

Comment (laushinka) [reply components/dashboard/src/teams/TeamUsage.tsx:98]: > I suspect the current logic has a bug for when 7 months back goes back to the previous year, as then you need to wrap the month and subtract 1 year

It actually computes it as expected! I tried going back 12 months from July 2022 and it correctly shows me as July 2021.

Comment (easyCZ) [reply components/dashboard/src/teams/TeamUsage.tsx:98]: I'm surprised! But the JS version of Date, which is based on Java 1.0 package actually allows you to subtract from dates by using negative months!

```
var now = new Date()
now
Wed Jul 27 2022 11:02:40 GMT+0200 (Central European Summer Time)

new Date(now.getFullYear(), now.getMonth() - 10, 0)
Tue Aug 31 2021 00:00:00 GMT+0200 (Central European Summer Time)
```

**Review by easyCZ — APPROVED** [review]
LGTM

/hold
In case you want someone else to review.

**Comment by laushinka** [comment]
> LGTM
> 
> /hold In case you want someone else to review.

Thanks! The functionality works but the list of months is much uglier than the dropdown. Since there are other things that needs to be fixed, I'll unhold and move on to not block subsequent work, but I'm aware that we'll need to implement the dropdown or at least improve how it looks. 
Again, really sorry @gtsiolis 🙏🏽
